web/src/compose_recipient.js

```diff
@@ -186,36 +198,42 @@ export function on_compose_select_recipient_update(new_value) {
         const $stream_header_colorblock = $(
             "#compose_recipient_selection_dropdown .stream_header_colorblock",
         );
-        stream_bar.decorate(new_value, $stream_header_colorblock);
-        if (message_type === "private") {
-            switch_message_type("stream");
-        }
+        const stream_name = compose_state.stream_name();
+        stream_bar.decorate(stream_name, $stream_header_colorblock);
+        switch_message_type("stream");
         // Always move focus to the topic input even if it's not empty,
         // since it's likely the user will want to update the topic
         // after updating the stream.
         ui_util.place_caret_at_end($("#stream_message_recipient_topic")[0]);
-        check_stream_posting_policy_for_compose_box(new_value);
+        check_stream_posting_policy_for_compose_box(stream_name);
```

Comment:
Just making a note for the potential follow-up of having a bunch of functions like this one pass around either stream IDs or `sub` objects depending on what makes sense.